langchain4j/src/main/java/dev/langchain4j/service/tool/DefaultToolExecutor.java

```diff
@@ -138,7 +138,8 @@ static Object coerceArgument(
             try {
                 @SuppressWarnings({"unchecked", "rawtypes"})
                 Class<Enum> enumClass = (Class<Enum>) parameterClass;
-                return Enum.valueOf(enumClass, Objects.requireNonNull(argument.toString()));
+                // Safe to assume that enum values are uppercased
+                return Enum.valueOf(enumClass, Objects.requireNonNull(argument.toString().toUpperCase()));
```

Comment:
I am not sure about this change, what if enum is not uppercase?
What about trying to parse it as-is first and if it fails, trying to uppercase it before giving up?

Reply:
That's fine.  The model wasn't upper-casing it and java convention is to uppercase

Reply:
done